internal/ignition/ignition.go

```diff
@@ -391,6 +391,12 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte,
 		}
 	}
 
+	err = g.applyManifestPatches(ctx)
+	if err != nil {
+		log.WithError(err).Errorf("failed to apply manifests' patches for cluster '%s'", g.cluster.ID)
```

Comment:
We probably want to clean up the mess just as we do above (also in the conditions below)

Maybe it's best to just defer a conditional cleanup when err != nil 

```suggestion
		_ = os.Remove(filepath.Join(g.workDir, "manifests"))
		_ = os.Remove(filepath.Join(g.workDir, "openshift"))
		log.WithError(err).Errorf("failed to apply manifests' patches for cluster '%s'", g.cluster.ID)
```

Reply:
mmh, I think we can add this too. I added a `Remove` in the `applyManifestPatches` function to remove the `.patch` files and prevent from injecting those in the ignition file. I think adding this remove (with a defer) here also makes sense since at the end of this call we should've already generated the ignition file.